src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/directives/processor_ui_convert.js

```diff
@@ -0,0 +1,46 @@
+import uiModules from 'ui/modules';
+import template from '../views/processor_ui_convert.html';
+import _ from 'lodash';
+import keysDeep from '../lib/keys_deep';
+
+const app = uiModules.get('kibana');
+
+//scope.processor, scope.pipeline are attached by the process_container.
+app.directive('processorUiConvert', function () {
+  return {
+    restrict: 'E',
+    template: template,
+    controller : function ($scope) {
+      const processor = $scope.processor;
+      const pipeline = $scope.pipeline;
+
+      function consumeNewInputObject() {
+        $scope.fields = keysDeep(processor.inputObject);
+        refreshFieldData();
+      }
+
+      function refreshFieldData() {
+        $scope.fieldData = _.get(processor.inputObject, processor.sourceField);
+      }
+
+      function processorUiChanged() {
+        pipeline.setDirty();
+      }
+
+      $scope.types = ['auto', 'number', 'string', 'boolean'];
+
+      $scope.$watch('processor.inputObject', consumeNewInputObject);
+
+      $scope.$watch('processor.sourceField', () => {
+        if (_.isEmpty($scope.processor.targetField)) {
+          $scope.processor.targetField = $scope.processor.sourceField;
+        }
```

Comment:
I think this comment got clobbered by an unrelated commit, reposting it:

Given that targetField is optional, let's not autofill it (and add a label saying it's optional). Current behavior is confusing because it gets autofilled on the initial selection of field, but then any changes to field leave targetField alone, which caused me to overwrite the first field without realizing it.